### ISSUE TITLE
fix(eslint-config): typescript doesn't catch the error about array spreading in parameters

### ISSUE THREAD

**Issue by zanminkian** [zanminkian/git-validator#244]
```ts
function arr() {
  return ['abc', 'def'];
}
const set = new Set(...arr); // should cause error here
```

**Comment by zanminkian** [comment]
https://github.com/microsoft/TypeScript/issues/59390

**Comment by zanminkian** [comment]
Waiting [eslint-plugin-unicorn](https://github.com/sindresorhus/eslint-plugin-unicorn/issues/2412) to fix it or create a rule to fix it.
